api-responses/make_api_responses.sh

```diff
@@ -0,0 +1,66 @@
+# The following script sets up two test nodes,
+# Test node 1: 2 OpenNeuro datasets, in the aggregate results mode
+# Test node 2: the (default) BIDS Synthetic dataset, in the unaggregated results mode
+#
+# A full_stack deployment recipe will be used for Test node 1 to provide local federation.
+
+mkdir test_node1 test_node2 api-responses
+
+# SETUP FOR TEST NODE 1
+git clone https://github.com/neurobagel/recipes.git test_node1/recipes
+
+# Get the data
+git clone https://github.com/neurobagel/openneuro-annotations.git test_node1/openneuro-annotations
+mkdir test_node1/openneuro_mini_data
+cp test_node1/openneuro-annotations/ds000001.jsonld test_node1/recipes/ds000001.jsonld test_node1/openneuro_mini_data
+
+cd test_node1/recipes
+cp template.env .env
+
+sed -i 's/^LOCAL_GRAPH_DATA=.*/LOCAL_GRAPH_DATA=../openneuro_mini_data/' .env
+sed -i 's/^NB_FEDERATE_REMOTE_PUBLIC_NODES=.*/NB_FEDERATE_REMOTE_PUBLIC_NODES=False/' .env
+sed -i 's/^NB_API_QUERY_URL=.*/NB_API_QUERY_URL=http://localhost:8080/' .env
+sed -i 's/^COMPOSE_PROFILES=.*/COMPOSE_PROFILES=full_stack/' .env
+
+# Use full container names to avoid conflicts
+echo '[{"NodeName": "2 OpenNeuro Datasets", "ApiURL": "http://neurobagel_node-api-1:8000"},{"NodeName": "BIDS Synthetic", "ApiURL": "http://neurobagel_node2-api-1:8000"}]' > local_nb_nodes.json
```

Comment:
Interesting, did you experience an error when using the service name? Or is this needed because the nodes are not on the same docker network?

Reply:
You mean just `api` alone? The issue is that in each node's recipe, the n-API always has the same Compose _service_, name `api`, so we need a way to deconflict the two n-APIs / make sure we know which `api` we're referring to once they're in the same network. To do that explicitly, here we're using the _container_ names that the n-APIs will have once running, which by default will have the Docker Compose project name prepended to it (and at this point we've manually changed the project name for one of the two nodes).

The n-APIs need to be in the same Docker network as the f-API for the f-API to be able to refer to them by either service name or container name. That's why we have this line in the script as well:

```
docker network add neurobagel_node_default neurobagel_node2-api-1
```